File: rust/nvim_tools/src/oxi/find.rs
use crate::types::dto::CmdResult;
use crate::types::dto::FindFilesParams;
use crate::util;

pub fn find_files(params: FindFilesParams) -> CmdResult<util::find::FindFilesSucceedResult> {
    let options = util::find::FindFilesOptions {
        workspace: params.workspace,
        cwd: params.cwd,
        flag_case_sensitive: params.flag_case_sensitive,
        flag_gitignore: params.flag_gitignore,
        flag_regex: params.flag_regex,
        search_pattern: params.search_pattern,
        search_paths: params.search_paths,
        exclude_patterns: params.exclude_patterns,
    };

    let cmd_result: CmdResult<util::find::FindFilesSucceedResult> = {
        let result = util::find::find_files(&options);
        match result {
            Ok(data) => CmdResult {
                cmd: data.cmd.to_owned(),
                error: None,
                data: Some(data),
            },
            Err(data) => CmdResult {
                cmd: data.cmd.to_owned(),
                error: Some(data.error),
                data: None,
            },
        }
    };
    cmd_result
}

File: lua/__types__/rstd/find.lua
---@meta

---@module 'rstd.find'
---@class rstd.find
local M = {}

---@class rstd.find.IFindFilesOptions
---@field public cwd                    string
---@field public flag_case_sensitive    boolean
---@field public flag_gitignore         boolean
---@field public flag_regex             boolean
---@field public search_pattern         string
---@field public search_paths           string
---@field public exclude_patterns       string

---@class rstd.find.IFindFilesSucceedResult
---@field public filepaths              string[]

---@class rstd.find.IFindFilesFailedResult
---@field public error                  string

---@param options                       rstd.find.IFindFilesOptions
---@return rstd.find.IFindFilesSucceedResult|nil
---@return rstd.find.IFindFilesFailedResult|nil
function M.find_files(options) end

return M
